addons/hr_expense/models/hr_expense.py

```diff
@@ -919,12 +919,11 @@ def _compute_amount(self):
         for sheet in self:
             sheet.total_amount = sum(sheet.expense_line_ids.mapped('total_amount_company'))
 
-    @api.depends("account_move_id.line_ids")
+    @api.depends('account_move_id.amount_residual_signed')
     def _compute_amount_residual(self):
         for sheet in self:
-            payment_term_lines = sheet.account_move_id.sudo().line_ids \
-                .filtered(lambda line: line.expense_id in sheet.expense_line_ids and line.account_internal_type in ('receivable', 'payable'))
-            sheet.amount_residual = -sum(payment_term_lines.mapped('amount_residual'))
+            # Expense moves are outbound, so amount_residual_signed are negative
+            sheet.amount_residual = -sheet.account_move_id.amount_residual_signed
```

Comment:
Not your fault, but it's weird that the `amount_residual` is in document currency in the accounting, but not for expenses, what do you think ? :grimacing:

Reply:
That's because expense reports moves in foreign currency are a very recent thing. And we still don't allow it (yet, finger-crossed) on expenses paid by employee.

Also, the expense lines could in later versions create a move for every expenses using the respective currency of each (when paid by company). Keeping an amount_residual in company currency makes more sense